sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -1268,12 +1273,18 @@ export class ApplicationService extends RecordDataModelService<Application> {
         applicationStatus: ApplicationStatus.completed,
       },
     };
-    if (applicationId) {
+    if (applicationNumber) {
```

Comment:
Existing bug which was fixed.

Reply:
Just as a suggestion, I believe that the below would also work if `applicationNumber` and `applicationId` are undefined in the method parameter. This seems to be one of the first that we did.
```ts
const findQuery: FindOneOptions<Application> = {
      select: {
        id: true,
        applicationNumber: true,
        isArchived: true,
      },
      where: {
        student: { user: { id: userId } },
        applicationStatus: ApplicationStatus.completed,
        applicationNumber,
        id: applicationId,
      },
    };
```

Reply:
what was the bug here?

Reply:
@ann-aot  the bug here is, we check for applicationId and put applicationNumber in the where clause.

Reply:
Thanks for doing the refactor 😉